### PR TITLE
Governance voting changes caused by boosted baln

### PR DESCRIPTION
**Pull request by simusud**
governance voting changes caused by boosted baln 
updated unit tests for governance voting

### CODE REVIEW

File: core-contracts/Governance/src/main/java/network/balanced/score/core/governance/GovernanceImpl.java [hunk 1]
```diff
@@ -198,7 +198,7 @@ public void cancelVote(BigInteger vote_index) {
         Context.require(Context.getCaller().equals(proposal.proposer.get()) ||
                         Context.getCaller().equals(Context.getOwner()),
                 "Only owner or proposer may call this method.");
-        if (proposal.startSnapshot.get().compareTo(getDay()) <= 0) {
+        if (proposal.startSnapshot.get().compareTo(BigInteger.valueOf(Context.getBlockHeight())) <= 0) {
```
Comment (AntonAndell) [inline core-contracts/Governance/src/main/java/network/balanced/score/core/governance/GovernanceImpl.java:201]: @nightowl121 wont bBaln adhere to timeOffset? If not perhaps we should add a setTimeOffset to bBaln. Could make some integration tests possible aswell

**Comment by AntonAndell** [comment]
We will also change the define vote signature in this PR https://github.com/balancednetwork/balanced-java-contracts/pull/162

So we could combine them if we wanna do any other changes to the function parameters

**Comment by dbrehmer** [comment]
Add forum link issue to this work #26 
See notes on issue #29 as well.

File: core-contracts/Governance/src/main/java/network/balanced/score/core/governance/GovernanceImpl.java [hunk 2]
```diff
@@ -349,7 +363,7 @@ public void evaluateVote(BigInteger vote_index) {
         BigInteger endSnap = proposal.endSnapshot.get();
         String actions = proposal.actions.get();
         BigInteger majority = proposal.majority.get();
-
+        BigInteger currentBlockHeight = BigInteger.valueOf(Context.getBlockHeight());
```
Comment (AntonAndell) [inline core-contracts/Governance/src/main/java/network/balanced/score/core/governance/GovernanceImpl.java:366]:
```suggestion
```

File: core-contracts/Governance/src/main/java/network/balanced/score/core/governance/ProposalDB.java [hunk 3]
```diff
@@ -88,6 +91,7 @@ public static ProposalDB createProposal(String name,
         newProposal.voteSnapshot.set(snapshot);
         newProposal.startSnapshot.set(start);
         newProposal.endSnapshot.set(end);
```
Comment (AntonAndell) [inline core-contracts/Governance/src/main/java/network/balanced/score/core/governance/ProposalDB.java:93]: Think it is time to rename these variables, something like
snapshotBlock
startDay
endDay
No need to change the strings keys in checkVote though

File: core-contracts/Governance/src/test/java/network/balanced/score/core/governance/GovernanceTestBase.java [hunk 4]
```diff
@@ -173,30 +175,31 @@ private void setupAddresses() {
...
-        when(baln.mock.stakedBalanceOf(owner.getAddress())).thenReturn(BigInteger.TEN.multiply(ICX));
-        
-        governance.invoke(owner, "defineVote", name, description, voteStart, snapshot, actions);
+        governance.invoke(owner, "defineVote", name, description, voteStart, BigInteger.TWO, forumLink, actions);
         BigInteger id = (BigInteger) governance.call("getVoteIndex", name);
 
-        when(baln.mock.totalStakedBalanceOfAt(snapshot)).thenReturn(BigInteger.valueOf(6).multiply(ICX));
-    
+        when(bBaln.mock.totalSupplyAt(any(BigInteger.class))).thenReturn(BigInteger.valueOf(6).multiply(ICX));
+
```
Comment (AntonAndell) [inline core-contracts/Governance/src/test/java/network/balanced/score/core/governance/GovernanceTestBase.java:192]:
```suggestion
```

File: core-contracts/Governance/src/test/java/network/balanced/score/core/governance/GovernanceVotingTest.java [hunk 5]
```diff
@@ -41,68 +40,64 @@ public class GovernanceVotingTest extends GovernanceTestBase {
...
         // Arrange
-        sm.getBlock().increase(DAY);
         Account accountWithLowBalance = sm.createAccount();
         BigInteger day = (BigInteger) governance.call("getDay");
         String name = "test";
+        String forumLink = "https://gov.balanced.network/";
         String description = "test vote";
         BigInteger voteStart = day.add(BigInteger.TWO);
-        BigInteger snapshot = day.add(BigInteger.ONE);
+        ;
```
Comment (AntonAndell) [inline core-contracts/Governance/src/test/java/network/balanced/score/core/governance/GovernanceVotingTest.java:55]:
```suggestion
```

File: core-contracts/Governance/src/test/java/network/balanced/score/core/governance/GovernanceVotingTest.java [hunk 6]
```diff
@@ -167,30 +162,29 @@ void cancelVote_Owner() {
...
-        sm.getBlock().increase(DAY);
         Account proposer = sm.createAccount();
         Account nonProposer = sm.createAccount();
         BigInteger day = (BigInteger) governance.call("getDay");
         String name = "test";
+        String forumLink = "https://gov.balanced.network/";
         String description = "test vote";
         BigInteger voteStart = day.add(BigInteger.TWO);
-        BigInteger snapshot = day.add(BigInteger.ONE);
+        ;
```
Comment (AntonAndell) [inline core-contracts/Governance/src/test/java/network/balanced/score/core/governance/GovernanceVotingTest.java:172]:
```suggestion
```

**Review by AntonAndell — CHANGES_REQUESTED** [review]
Looks good, just some small changes then it is probably fine to merge into the bBaln branch

**Review by nightowl121 — APPROVED** [review]
Approved

**Pull request merged**